@dependents ignores customized task name

```python
@task("task_3")
@depends(task1)
@dependents(task2)
def task3():
     pass
```

Would produce `BUILD FAILED - Missing task 'task3' required for task 'task2'`

--- a/src/main/python/pybuilder/reactor.py
+++ b/src/main/python/pybuilder/reactor.py
@@ -219,6 +219,7 @@ class Reactor(object):
 
         for name in dir(project_module):
             candidate = getattr(project_module, name)
+            name = normalize_candidate_name(candidate)
 
             if getattr(candidate, TASK_ATTRIBUTE, None):
                 dependents = getattr(candidate, DEPENDENTS_ATTRIBUTE, None)
@@ -228,9 +229,9 @@ class Reactor(object):
                     for d in dependents:
                         if isinstance(d, optional):
                             d = d()
-                            add_task_dependency(d, candidate, True)
+                            add_task_dependency(d, name, True)
                         else:
-                            add_task_dependency(d, candidate, False)
+                            add_task_dependency(d, name, False)
 
         for name in dir(project_module):
             candidate = getattr(project_module, name)

--- a/src/unittest/python/reactor_tests.py
+++ b/src/unittest/python/reactor_tests.py
@@ -202,6 +202,41 @@ class ReactorTest(unittest.TestCase):
             self.execution_manager.register_late_task_dependencies.assert_has_calls(
                 [call({}), call({"task2": [TaskDependency(task3)]})])
 
+    def test_task_dependencies_with_post_definition_injections_custom_names(self):
+        import pybuilder.reactor
+
+        with patch("pybuilder.reactor.Task"):
+            @task
+            def task1():
+                pass
+
+            @task
+            @depends(task1)
+            def task2():
+                pass
+
+            @task("task_3")
+            @depends(task1)
+            @dependents(task2)
+            def task3():
+                pass
+
+            module1 = ModuleType("mock_module_one")
+            module1.task1 = task1
+            module1.task2 = task2
+
+            module2 = ModuleType("mock_module_two")
+            module2.task3 = task3
+
+            self.reactor.collect_tasks_and_actions_and_initializers(module1)
+            pybuilder.reactor.Task.assert_has_calls([call("task1", task1, [], ''),
+                                                     call("task2", task2, [TaskDependency(task1)], '')])
+
+            self.reactor.collect_tasks_and_actions_and_initializers(module2)
+            pybuilder.reactor.Task.assert_has_calls([call("task_3", task3, [TaskDependency(task1)], '')])
+            self.execution_manager.register_late_task_dependencies.assert_has_calls(
+                [call({}), call({"task2": [TaskDependency("task_3")]})])
+
     def test_should_collect_single_before_action(self):
         @before("spam")
         def action():

